obfs4/obfs4.go

```diff
@@ -23,6 +23,24 @@ import (
 	"golang.org/x/net/proxy"
 )
 
+type DialFunc func(string, string) (net.Conn, error)
```

Comment:
Please, document new public types. Are not using a context-enabled dialer because other projects with which you want to be compatible do not define a context-enabled dialer as well?

Reply:
no, I think this was mostly because `obfs4` passes a `DialFunc`. when integrating in OONI probably reusing `OBFS4Dialler`  (with a `DialContext` method) would be the right thing to do...

Reply:
I've been looking at this again, and I think the reason I avoided going that way is that it entails quite a bit of code duplication (obfs4 transport).

Reply:
Did another take, I was wrong about the code duplication (`probe-cli` does, but I don't seem to need it here). I think I've addressed the context-enabled dialer problem now.